src/main/java/com/internet/shop/dao/ShoppingCartDao.java

```diff
@@ -0,0 +1,21 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.model.Product;
+import com.internet.shop.model.ShoppingCart;
+import java.util.Optional;
+
+public interface ShoppingCartDao {
+    ShoppingCart create(ShoppingCart shoppingCart);
+
+    ShoppingCart addProduct(ShoppingCart shoppingCart, Product product);
+
+    boolean deleteProduct(ShoppingCart shoppingCart, Product product);
+
+    void clear(ShoppingCart shoppingCart);
```

Comment:
these methods should be at the service layer

Reply:
Thx. Fixed.